executor/aggfuncs/func_bit_or.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type bitOrUint64 struct {
+	baseAggFunc
+}
+
+type partialResult4BitFunc struct {
+	value uint64
+}
+
+func (e *bitOrUint64) AllocPartialResult() PartialResult {
+	return PartialResult(&partialResult4BitFunc{})
+}
+
+func (e *bitOrUint64) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4BitFunc)(pr)
+	p.value = 0
+}
+
+func (e *bitOrUint64) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4BitFunc)(pr)
+	chk.AppendUint64(e.ordinal, p.value)
+	return nil
+}
+
+func (e *bitOrUint64) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
+	p := (*partialResult4BitFunc)(pr)
+	for _, row := range rowsInGroup {
+		inputValue, isNull, err := e.args[0].EvalInt(sctx, row)
```

Comment:
Should we wrap a cast as uint in `typeInfer4BitFuncs`,
or bit_or(varchar) may fail?

Reply:
bit_or( varchar ) will return 0.

Reply:
actually, we need to add a cast, consider this case:
```sql
drop table if exists t;
create table t(a decimal(10, 4));
insert into t values(12.2);
select bit_or(a) from (select * from t union all select * from t) tmp;
```
```sql
TiDB(localhost:4000) > desc select bit_or(a) from (select * from t union all select * from t) tmp;
+--------------------------+------+----------------------------------------------+----------+
| id                       | task | operator info                                | count    |
+--------------------------+------+----------------------------------------------+----------+
| StreamAgg_13             | root | funcs:bit_or(tmp.a)                          | 1.00     |
| └─Union_21               | root |                                              | 20000.00 |
|   ├─TableReader_24       | root | data:TableScan_23                            | 10000.00 |
|   │ └─TableScan_23       | cop  | table:t, range:[-inf,+inf], keep order:false | 10000.00 |
|   └─TableReader_27       | root | data:TableScan_26                            | 10000.00 |
|     └─TableScan_26       | cop  | table:t, range:[-inf,+inf], keep order:false | 10000.00 |
+--------------------------+------+----------------------------------------------+----------+
6 rows in set (0.00 sec)
```

The above `StreamAgg_13` directly handles the original data instead of another aggregate operator's partial result, which is guaranteed to be uint64. This PR may failed on this query if we don't wrap a cast on it's parameter.

Reply:
Ye, I'll fix it